src/vs/workbench/browser/viewlet.ts

```diff
@@ -173,6 +177,16 @@ export class ViewletRegistry extends CompositeRegistry<Viewlet> {
 	}
 
 	/**
+	 * Registers multiple external viewlets.
+	 */
+	public registerExternalViewlets(descriptors: ViewletDescriptor[]): void {
```

Comment:
Yes, I also found this weird when initially scanning the PR and suggest to fold this into the same solution we already have. There is no need to distinguish external and "internal" viewlets if possible.